src/libcrun/linux.c

```diff
@@ -3789,36 +3898,138 @@ prepare_and_send_mounts (libcrun_container_t *container, pid_t pid, int sync_soc
         how_many++;
     }
 
-  ret = TEMP_FAILURE_RETRY (write (sync_socket_host, &how_many, sizeof (how_many)));
+  return send_mounts (sync_socket_host, mount_fds, how_many, def->mounts_len, err);
+}
+
+static int
+prepare_and_send_dev_mounts (libcrun_container_t *container, int sync_socket_host, libcrun_error_t *err)
+{
+  runtime_spec_schema_config_schema *def = container->container_def;
+  cleanup_close_map struct libcrun_fd_map *dev_fds = NULL;
+  bool has_userns = (get_private_data (container)->unshare_flags & CLONE_NEWUSER) ? true : false;
+  cleanup_close int current_mountns = -1;
+  cleanup_free char *state_dir = NULL;
+  cleanup_free char *devs_path = NULL;
+  cleanup_close int devs_mountfd = -1;
+  cleanup_close int targetfd = -1;
+  size_t how_many = 0;
```

Comment:
nit: should we rename s/how_many/mount_count or to something similar ( not a blocker )

Reply:
I've reused the same name used by `send_mounts` but I can rename if it is clearer

Reply:
no not necessarily :) I think its okay if this is a consistent name across the code I just had to skim lines below to check that this is a counter ( but this is not a big deal I think if we change this then maybe across all the files so I think a different PR maybe ? )